lib/seahorn/BvOpSem2.cc

```diff
@@ -774,6 +797,55 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
 
   void visitBranchSentinel(CallSite CS) {}
 
+  void visitGetShadowMem(CallSite CS) {
+    if (!m_ctx.getMemReadRegister()) {
+      LOG("opsem", ERR << "No read register found - check if corresponding"
+                          "shadow instruction is present.");
+      m_ctx.setMemReadRegister(Expr());
+      return;
+    }
+    Expr slot = lookup(*CS.getArgument(0));
+    if (!m_ctx.alu().isNum(slot)) {
+      LOG("opsem", ERR << "Metadata slot should resolve to a number.");
+      assert(false);
+    }
+    size_t slotNum = m_ctx.alu().toNum(slot).get_ui();
+    if (slotNum >= m_ctx.mem().getNumOfMetadataSlots()) {
+      LOG("opsem", ERR << "Metadata slot exceeds number of available slots.");
+      assert(false);
```

Comment:
`assert` only works in debug build I think, so is this the desired behaviour in other build types? or should we use `llvm::unreacheable`?

Reply:
I think `llvm::unreachable` has undefined behavior and should not be used to describe an exception

https://llvm.org/docs/LangRef.html#unreachable-instruction